cartoframes/magic.py

```diff
@@ -0,0 +1,82 @@
+"""carto magics"""
+from IPython.core.magic import (Magics, magics_class, line_magic,
+                                cell_magic, line_cell_magic)
+from cartoframes.context import CartoContext
+from cartoframes import utils
+# GOALS
+# - have it automatically find a CartoContext object
+# - make it agnostic to line or cell magics
+# - maps!
+
+def null_to_none(string):
+    """changes null to None"""
+    return None
+
+@magics_class
+class CartoMagics(Magics):
+    """Magics for bringing CARTO exploratory data analysis to IPython"""
+    @line_cell_magic
+    def carto(self, line, cell=None):
```

Comment:
We should maybe find a better name for this magic.. how about `cartodata` or `cartoread`? I like the second better because `cartodata` could be misconstrued as `cc.data`.

Reply:
cartoread is good! or even cartoquery... because you can pass in SQL statements in the cell